src/pages/saksbehandling/steg/beregning/FradragInputs.tsx

```diff
@@ -0,0 +1,290 @@
+            {Object.values(Fradragstype).map((f) => (
+                <option value={f} key={f}>
+                    {props.intl.formatMessage({ id: fradragstypeResourceId(f) })}
+                </option>
+            ))}
+        </Select>
+        <Feilmelding>{props.feil ? props.feil : ''}</Feilmelding>
+    </div>
+);
+
+const fraUtlandInntekt = yup
+    .object<FraUtlandInntektFormData>()
+    .defined()
+    .when('fraUtland', {
+        is: true,
+        then: yup.object<FraUtlandInntektFormData>({
+            beløpUtenlandskValuta: validateStringAsNumber,
+            valuta: yup.string().required(),
+            kurs: validateStringAsNumber,
+        }),
+        otherwise: yup.object<FraUtlandInntektFormData>(),
+    });
+
+const delerAvPeriode = yup
+    .object<DelerAvPeriodeFormData>()
+    .defined()
+    .when('delerAvPeriodeChecked', {
+        is: true,
+        then: yup.object<DelerAvPeriodeFormData>({
+            fraOgMed: yup.date().required().nullable(),
+            tilOgMed: yup.date().required().nullable(),
+        }),
+        otherwise: yup.object<DelerAvPeriodeFormData>(),
+    });
+
+export const fradragSchema = yup.object<FradragFormData>({
+    beløp: validateStringAsNumber,
+    type: yup.string().defined().oneOf(Object.values(Fradragstype), 'Du må velge en fradragstype'),
+    fraUtland: yup.boolean(),
+    fraUtlandInntekt: fraUtlandInntekt,
+    delerAvPeriodeChecked: yup.boolean(),
+    delerAvPeriode: delerAvPeriode,
+});
+
+export const isValidFradrag = (f: unknown): f is Fradrag => fradragSchema.isValidSync(f);
+
+export const FradragInputs = (props: {
+    fradrag: Array<FradragFormData>;
+    feltnavn: string;
+    errors: string | string[] | FormikErrors<FradragFormData>[] | undefined;
+    intl: IntlShape;
+    onChange: (e: React.ChangeEvent<unknown>) => void;
+    setFieldValue: (field: string, value: Date | [Date, Date] | null | string) => void;
+    onLeggTilClick: () => void;
+    onFjernClick: (index: number) => void;
+}) => {
+    return (
+        <div className={styles.fradragContainer}>
+            {typeof props.errors === 'string' && props.errors}
+
+            {props.fradrag.map((fradrag, index) => {
+                const errorForLinje = Array.isArray(props.errors) ? props.errors[index] : null;
+                const name = `${props.feltnavn}[${index}]`;
+                const typeId = `${name}.type`;
+                const belopId = `${name}.beløp`;
+                const fraUtlandId = `${name}.fraUtland`;
+                const beløpUtenlandskValutaId = `${name}.fraUtlandInntekt.beløpUtenlandskValuta`;
+                const valutaId = `${name}.fraUtlandInntekt.valuta`;
+                const kursId = `${name}.fraUtlandInntekt.kurs`;
+                const delerAvPeriodeId = `${name}.delerAvPeriodeChecked`;
+                const fraOgMedId = `${name}.delerAvPeriode.fraOgMed`;
+                const tilOgMedId = `${name}.delerAvPeriode.tilOgMed`;
```

Comment:
Tänker att upplägget med att man skal sende in ett feltnavn och sen ändra en specifik key till det feltnavnet er ganska error-prone och ovanligt. Exempelvis på linje 152, hvordan vet man att `${name}` faktiskt har `fraUtlandInntekt`?

Reply:
Fra det jeg forstår, så 'vet' man ikke at man har fraUtlandInntekt. Den blir opprettet av formik med det navnet den får. 

Er enig med at det kan være error prone. Skal ta en titt på det :)

Reply:
@aryani1 Det er vel gjort sånn her for å enkelt kunne bruke `formik` sin `handleChange`-funksjon. Enig i at det er error prone. Ser det finnes et par issues som diskuterer ting som dette; https://github.com/microsoft/TypeScript/issues/12290, https://github.com/microsoft/TypeScript/pull/40336. Hadde vært spennende å se om vi kunne bruke noe fra dem for å få typet det bedre.